RPGManager_development/test/unit_test/character_sheet_unit_tests/ArmorsUnitTests.java

```diff
@@ -0,0 +1,150 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.assertEquals;
+
+import java.security.InvalidParameterException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.Armors;
+import rpg_database.character_sheet.BaseClasses;
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+
+public class ArmorsUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetArmorsMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Armors");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, malformedInput);
+	}
+
+	// default Armors for BaseClasses
+
+	@Test
+	public void testGetDefaultArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.HEAVY_LEATHER, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForRogue() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.ROGUE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_LEATHER, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForMage() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.MAGE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.ROBE, actualValue);
+	}
+
+	// default ArmorRating for BaseClasses
+
+	@Test
+	public void testGetDefaultArmorRating() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		int actualArmorValue = characterSheet.getData(Fields.ARMOR_RATING);
+		assertEquals(0, actualArmorValue);

```

Comment:
Szóval reflektálva arra hogy ugye jelenleg alapjáratban 0-ra állítja defaultban a karakterlap az armorvalue értékét, amit feljebb említettél. 2 lehetőség van:
- vagy megoldjuk, hogy le legyen ez a kivétel kezelve, mondjuk inicializálásnál egy olyasmival, hogy:
```java
private void putDefaultValueByFieldAllowedType(Fields field) {
		Class<?> allowedType = field.getAllowedClass();
		Object defaultValue;
		if (allowedType == CharacterAttribute.class) {
			defaultValue = new CharacterAttribute(0, BaseClasses.WARRIOR.isAttributeMajor(field));
		} else if (field.name().equals("ARMOR_RATING")) {
                        defaultValue = 4;
                } else {
			defaultValue = defaultData.get(allowedType);
		}
		characterData.put(field, allowedType.cast(defaultValue));
	}
```

- vagy azt is lehet hogy default esetben nem adunk armor type-ot ami ugye igazolná a nullás értéket. Persze akkor ennek az eseteit - hogy milyen módon van értelmezve az hogy egy karakteren nincs armor - kezelni kell. Vizsgálni hogy az az érték null -e stb.
- ultra slendrián megoldás: alapjáratban ROBE-ot adsz a karakterre. 0-s értéke van és alig kell valamit átírni. Persze nem elegáns... és nem is a kedvenc opcióm. Beszéljünk róla hogy ezek közül te mit támogatnál

És persze ezekhez mérten változhat itt a teszt is ugye.